specs/popup.md

```diff
@@ -0,0 +1,136 @@
+
+## Position and Collision
+
+* Position - The point and pivot for popup to anchor alignment
+  * Anchor Position - Horizontal:
+    * left - the left point of the anchor element
+    * center - the center point of the anchor element
+    * right - the right point of the anchor element
+  * Anchor Position - Vertical:
+    * top - the top point of the anchor element
+    * center - the center point of the anchor element
+    * bottom - the bottom point of the anchor element
+  * Popup Position - Horizontal:
+    * left - the left point of the popup element
+    * center - the center point of the popup element
+    * right - the right point of the popup element
+  * Popup Position - Vertical:
+    * top - the top point of the popup element
+    * center - the center point of the popup element
+    * bottom - the bottom point of the popup element
+
+### Positioning
+
+* Points of reference
+
+    There are 9 points on the anchor and popup element to be used as reference to where the popup will open. The chosen anchor point will be the location on which the popup will pivot. The popup point will be the location on the popup element which will be anchored to the point on the anchor element.
+
+    ![Image of anchor positioning](./assets/popup/points.png)
+
+* Anchor default
+
+    ![Image of anchor positioning](./assets/popup/anchordefault.png)
+
+  * Anchor { vertical: 'bottom', horizontal: 'left'}
+  * Popup { vertical: 'top', horizontal: 'left'}
+
+* Anchor sample position
+
+    ![Image of anchor positioning](./assets/popup/anchorright.png)
+
+  * Anchor { vertical: 'bottom', horizontal: 'right'}
+  * Popup { vertical: 'top', horizontal: 'left'}
+
+* Anchor sample position
+
+    ![Image of anchor positioning](./assets/popup/anchorcenter.png)
+
+  * Anchor { vertical: 'center', horizontal: 'center'}
+  * Popup { vertical: 'top', horizontal: 'left'}
+
+* Popup sample position
+
+    ![Image of popup positioning](./assets/popup/popupbottom.png)
+
+  * Anchor { vertical: 'bottom', horizontal: 'left'}
+  * Popup { vertical: 'bottom', horizontal: 'left'}
+
+* Popup sample position
+
+    ![Image of popup positioning](./assets/popup/popupright.png)
+
+  * Anchor { vertical: 'bottom', horizontal: 'left'}
+  * Popup { vertical: 'top, horizontal: 'right}
+
+
+* Collision - The behavior of the popup when viewport limitations collide with the the position property
+  * Horizontal Collision:
+    * flip - Checks if the opposite side of the anchor has more space and if so flips it.
+    * fit - Shifts the popup away from the edge of the viewport until it has enough space or reaches other side of viewport
+    * flipfit - Applies flip logic and then fit logic
+    * none - no collision detection
+  * Vertical Collision - same as horizontal
+
+## Properties
+
+| Name | Type | Default | Required | Description |
+| -- | -- | -- | -- | -- |
+| anchor | Element\| Node | none | Yes | The element to be used as an anchor for the popup (will open next to it). |
```

Comment:
Suggestion: `Node | Point`. Element is an extension of Node, so you don't have to say that explicitly.